### PR TITLE
Switch to kustomize based rendering

### PR DESCRIPTION
**Pull request by glrf**
This PR switches from a manual patching of upstream resources to a kustomize based setup.

The diff is a mess, as the indentation for arrays changes for multiple resources. The following diff is a bit more useful:
https://github.com/projectsyn/component-kyverno/pull/67/files/8b39f89f33a6b16e17b25134096c0af15d6cbe42
(The two commits for this will be removed before merging)

This brings a few new clusterroles, changes some labels, and brings a minor config change, which AFAICT doesn't change much.

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: class/kyverno.yml [hunk 1]
```diff
@@ -1,73 +1,47 @@
...
+          - ${_kustomize_wrapper}
+        env_vars:
+          INPUT_DIR: ${_base_directory}/kyverno
+        args:
+          - \${compiled_target_dir}/${_instance}/kyverno
+
+      # Rename all files with a ":".
+      # This is necessary as otherwise commodore will not be able to render the diff.
+      # Caused by a bug in gitpython 3.1.24.
+      # Can be removed one gitpython is update to 3.1.28
```
Comment (bastjan) [inline class/kyverno.yml:27]:
```suggestion
      # Can be removed once gitpython is update to 3.1.28
```

**Review by simu — APPROVED** [review]
LGTM

File: component/kyverno.jsonnet [hunk 2]
```diff
@@ -37,85 +25,79 @@ local nodeSelectorConfig(role) =
...
+        count: params.replicas,
+      },
+    ],
+    patchesStrategicMerge: [
+      'deployment.yaml',
+      'namespace.yaml',
+    ],
+  },
+) {
+  // We need to patch namespace "kyverno" as the namespace renaming is only applied after doing strategic merge patches
```
Comment (bastjan) [inline component/kyverno.jsonnet:65]: Moving this comment to before the `namespace:` key would make it easier to read and cross-reference.

**Review by bastjan — APPROVED** [review]

File: component/kyverno.jsonnet (hunk 2 above)
Comment (glrf) [reply component/kyverno.jsonnet:74]: Ah yes you're right, that was moved to useless position :sweat_smile:

**Pull request merged**